ADAL/src/ADAuthenticationResult+Internal.m

```diff
@@ -175,20 +176,36 @@ + (ADAuthenticationResult*)resultForBrokerErrorResponse:(NSDictionary*)response
         protocolCode = [response valueForKey:@"code"];
     }
     
-    if (![NSString adIsStringNilOrBlank:protocolCode])
+    // Create error object according to error_domain value.
+    // Old version of broker won't send this value, we set it to ADAuthenticationErrorDomain like before
+    NSString *errorDomain = [response valueForKey:@"error_domain"] ? [response valueForKey:@"error_domain"] : ADAuthenticationErrorDomain;
+    if ([errorDomain isEqualToString:ADKeychainErrorDomain])
+    {
+        error = [ADAuthenticationError keychainErrorFromOperation:@"Broker keychain access" status:(OSStatus)errorCode correlationId:correlationId];
```

Comment:
can we just use a generic ADAuthenticationError method and pass it a domain? Then have special handling only for http headers? Having to map each of the domains is not very extensible.

Reply:
means we are going to declare [ADAuthenticationError errorWithDomainInternal:] public?

Reply:
Can't we use this?

+ (ADAuthenticationError *)errorWithDomain:(NSErrorDomain)domain
                                      code:(NSInteger)code
                              errorDetails:(NSString *)errorDetails
                             correlationId:(NSUUID *)correlationId;

We can extend it if necessary :)